pkg/dependencydiff_result.go

```diff
@@ -80,10 +85,100 @@ type DependencyCheckResult struct {
 	Name string `json:"name"`
 }
 
-// AsJSON for DependencyCheckResult exports the DependencyCheckResult as a JSON object.
-func (dr *DependencyCheckResult) AsJSON(writer io.Writer) error {
-	if err := json.NewEncoder(writer).Encode(*dr); err != nil {
-		return sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("encoder.Encode: %v", err))
+// FormatDependencydiffResults formats dependencydiff results.
+func FormatDependencydiffResults(
+	opts *options.Options,
+	depdiffResults []DependencyCheckResult,
+	doc checks.Doc,
+) error {
+	var err error
+
+	switch opts.Format {
+	case options.FormatDefault:
+		err = DependencydiffResultsAsString(depdiffResults, opts.ShowDetails, doc, os.Stdout)
+	case options.FormatJSON:
+		err = DependencydiffResultsAsJSON(depdiffResults, opts.ShowDetails, log.ParseLevel(opts.LogLevel), doc, os.Stdout)
+	case options.FormatMarkdown:
+		err = DependencydiffResultsAsMarkdown(depdiffResults, opts.Dependencydiff, doc, os.Stdout)
+	default:
+		err = sce.WithMessage(
+			sce.ErrScorecardInternal,
+			fmt.Sprintf(
+				"invalid format flag: %v. Expected [default, json, markdown]",
+				opts.Format,
+			),
+		)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to output dependencydiff results: %w", err)
+	}
+	return nil
+}
```

Comment:
Repeating my question from the design doc - is it worth adding support for and maintaining so many output formats for `depdiff`? Couldn't we instead assume JSON as the default output and skip the `--format` flag altogether?

Reply:
I also like not having a ton of options to support :)

Reply:
For the default format, my intention is to give the users a table like the default scorecard check results, this way it might be easier for users to read the results (and JSON might be more friendly to programming calls).

The markdown is prepared for the Action to use. I agree it could be a bit redundant here.

Reply:
about the supported output format:

removing the markdown support, which is an unnecessary one for now

we don't have doubts on JSON, and it could be the default output format for `dependencydiff`.

For the default format (the plaintext table one), should we still leave it for users to use?

my justification: 
Since users can specify the base and head using the branch names such as `main...dev` to see dep changes in the dev branch (the API will handle the branch name and take the head commit of that branch to use), a user might want to use this CLI in the terminal and see the default result which is more human-readable than JSON.

cons of introducing this extra format:
the overhead maintenance: Once it's added someone will depend on it and we'll have to maintain it in the long run

Is my justification a strong reason? @laurentsimon @naveensrinivasan  wdut?

Reply:
I don't fully follow the reasoning. My 2ct: JSON is easy to read, especially when piped into `jq` you have colors. The table always has problem due to terminal size. Nobody has been able to fix it till now.

Reply:
Thanks for your opinions! I'm inclined to only preserve JSON as the output format in v0, and we might discuss about this later whether we would add more format support to it.